pilot/cmd/pilot-agent/status/server.go

```diff
@@ -542,24 +571,62 @@ func negotiateMetricsFormat(contentType string) expfmt.Format {
 	return expfmt.FmtText
 }
 
-func processMetrics(metrics []byte) []byte {
-	return bytes.ReplaceAll(metrics, []byte("\n\n"), []byte("\n"))
+func copyAndProcessMetrics(dst io.Writer, src io.Reader, buf []byte) (written int64, err error) {
+	var sideBreak bool
+	for {
+		nr, er := src.Read(buf)
+		if nr > 0 {
+			rbuf := bytes.ReplaceAll(buf[0:nr], []byte("\n\n"), []byte("\n"))
```

Comment:
So essentially we are trying to copy everything from src to dst, utilizing `buf` along the way, and replacing `\n\n` with `\n`, yes?

A couple concerns here: 
* Its pretty complex. https://pkg.go.dev/strings#NewReplacer is something *really* close to this, but operates on strings. I wonder if we can somehow reuse some of that for our puroses?
* Its error prone. For this type of low level parsing stuff I think we absolutely must have fuzzing. See https://github.com/istio/istio/blob/master/pkg/fuzz/README.md for more info.


On the fuzzing in particular, I wrote up a first draft since I wanted to check some of my assumptions:

```go

func FuzzWrite(f *testing.F) {
	f.Fuzz(func(t *testing.T, d []byte, size uint8) {
		if size == 0 {
			return
		}
		if bytes.Contains(d, []byte("\n\n\n")) {
			// Doesn't work yet
			return
		}
		if bytes.HasSuffix(d, []byte("\n")) {
			// Doesn't work yet
			return
		}
		t.Logf("input: %v", d)
		buf := make([]byte, size)
		t.Logf("buffer size: %v/%v", len(buf), size)
		writer := &bytes.Buffer{}
		reader := bytes.NewReader(d)
		w, err := copyAndProcessMetrics(writer, reader, buf)
		if err != nil {
			t.Fatalf("got err: %v", err)
		}
		t.Logf("output: %v", writer.Bytes())

		if int(w) != (len(d) - bytes.Count(d, []byte("\n\n"))) {
			t.Fatalf("didn't write all: %v vs %v", w, len(d))
		}
		if bytes.Contains(writer.Bytes(), []byte("\n\n")) {
			t.Fatalf("contains double newline: %v", writer.String())
		}
	})
}
```

This exposes a few different issues, in particular a crash in some cases, and issues where the `\n\n` is split across the `buf` boundary (ex: buf size is 100, we have `\n\n` at index 100 and 101).